store/tikv/lock_resolver.go

```diff
@@ -363,6 +363,11 @@ func (lr *LockResolver) resolveLocks(bo *Backoffer, callerStartTS uint64, locks
 
 			if status.primaryLock != nil && status.primaryLock.UseAsyncCommit && !exists {
 				err = lr.resolveLockAsync(bo, l, status)
+				// Treat nonAsyncCommitLock error as a soft error. It happens when a transaction
+				// just fallbacks from async commit, so a simple retry will work.
+				if _, ok := err.(*nonAsyncCommitLock); ok {
+					msBeforeTxnExpired.update(0)
```

Comment:
It seems to be duplicated with `L377` below?

Reply:
The error should be ignore. Fixed.